src/module/mixer_4ch.rs

```diff
@@ -0,0 +1,65 @@
+use mixlab_protocol::{Mixer4chParams, LineType};
+
+use crate::engine::Sample;
+use crate::module::Module;
+
+#[derive(Debug)]
+pub struct Mixer4ch {
+    params: Mixer4chParams,
+}
+
+impl Module for Mixer4ch {
+    type Params = Mixer4chParams;
+    type Indication = ();
+
+    fn create(params: Self::Params) -> (Self, Self::Indication) {
+        (Mixer4ch { params }, ())
+    }
+
+    fn params(&self) -> Self::Params {
+        self.params.clone()
+    }
+
+    fn update(&mut self, params: Self::Params) -> Option<Self::Indication> {
+        self.params = params;
+        None
+    }
+
+    fn run_tick(&mut self, _t: u64, inputs: &[Option<&[Sample]>], outputs: &mut [&mut [Sample]]) -> Option<Self::Indication> {
+        let len = outputs[0].len();
+
+        for i in 0..len {
+            outputs[0][i] = 0.0;
```

Comment:
what is the value here if you don't `0.0` it?

Reply:
Currently it will be zero but that's only because we currently allocate fresh buffers on every tick. I'll move to reusing buffers at some point for perf and we'll need to make sure buffers are zeroed in that case. This is only an issue so far for 4ch mixer because it's the only module that might not output anything (in the case that all the inputs are disconnected)